### PR TITLE
Refactor and tighten up code

### PR DESCRIPTION
**Pull request by buddhamagnet**
1. Slim down main code by collapsing `raml.Method` checks into a map and loop.
2. Re-order exported and unexported functions for readability.
3. Drop some redundant test code and import `ramlapi` package into the test.
4. Strip down variable names and let the context do the talking.

### CODE REVIEW

File: ramlapi.go [hunk 1]
```diff
@@ -3,73 +3,40 @@ package ramlapi
...
-	}
-
-	// Get all children.
-	for nestname, nested := range resource.Nested {
-		return processResource(resourcepath, nestname, nested, routerFunc)
+// Process processes a RAML file and returns an API definition.
+func Process(file string) (*raml.APIDefinition, error) {
+	routes, err := raml.ParseFile(file)
+	if err != nil {
+		return nil, fmt.Errorf("Failed parsing RAML file: %s\n", err.Error())
```
Comment (ardan-bkennedy) [inline ramlapi.go]: Don't need to call Error(). The function will see the error interface is being implemented.

File: ramlapi.go [hunk 2]
```diff
@@ -82,31 +49,64 @@ func ResourceVerbs(resource *raml.Resource) map[string]map[string]string {
 			}
 		}
 	}
-	if resource.Post != nil {
-		verbs["POST"] = map[string]string{
-			"handler": resource.Post.DisplayName,
-		}
+
+	mappings := map[string]*raml.Method{
```
Comment (ardan-bkennedy) [inline ramlapi.go:53]: Just looking at the use of the pointer here. I need to know more about this type.

File: ramlapi.go [hunk 3]
```diff
@@ -82,31 +49,64 @@ func ResourceVerbs(resource *raml.Resource) map[string]map[string]string {
...
+			data["query_pattern"] = details["query_pattern"]
 		}
+		fun(data)
 	}
-	if resource.Delete != nil {
-		verbs["DELETE"] = map[string]string{
-			"handler": resource.Delete.DisplayName,
-		}
+
+	// Get all children.
```
Comment (ardan-bkennedy) [inline ramlapi.go]: Complete sentences in the comments.

File: ramlapi.go (hunk 2 above)
Comment (buddhamagnet) [inline ramlapi.go:53]: It's from the `raml` package which we import.

File: ramlapi.go (hunk 3 above)
Comment (buddhamagnet) [inline ramlapi.go]: That looks like a complete sentence to me, I didn't write it, but amended.

File: ramlapi.go (hunk 1 above)
Comment (buddhamagnet) [inline ramlapi.go]: Useful, thanks.

**Pull request closed without merging**